src/components/src/AppModalLogin/appmodallogin.main.tsx

```diff
@@ -19,232 +19,174 @@
  *
  */
 
-import React, { Component } from 'react';
+import React, { Component, useState, useEffect } from 'react';
 import { Link } from 'react-router-dom';
 import Modal from 'react-responsive-modal';
+import { FormikErrors, useFormik } from 'formik';
 import queryString from 'query-string';
 import { loginRegistered, loginRegisteredAuthService } from '../utils/AuthService';
-import './appmodallogin.main.less';
 import { getConfig, IEpConfig } from '../utils/ConfigProvider';
 
+import './appmodallogin.main.less';
+
 let Config: IEpConfig | any = {};
 let intl = { get: str => str };
 
 interface AppModalLoginMainProps {
   /** handle modal close */
-    handleModalClose: (...args: any[]) => any,
+  handleModalClose: (...args: any[]) => any,
   /** handle open modal */
-    openModal: boolean,
+  openModal: boolean,
   /** handle login */
-    onLogin?: (...args: any[]) => any,
+  onLogin?: (...args: any[]) => any,
   /** handle reset password */
-    onResetPassword?: (...args: any[]) => any,
+  onResetPassword?: (...args: any[]) => any,
   /** location search data */
-    locationSearchData?: string,
+  locationSearchData?: string,
   /** location path name */
-    locationPathName?: string,
+  locationPathName?: string,
   /** links for app modal login */
-    appModalLoginLinks: {
-        [key: string]: any
-    },
+  appModalLoginLinks: {
+    [key: string]: any
+  },
   /** show forgot password link */
-    showForgotPasswordLink: boolean,
+  showForgotPasswordLink: boolean,
   /** disable login */
-    disableLogin?: boolean,
+  disableLogin?: boolean,
   /** Open Id Connect Parameters */
-    oidcParameters?: {
-      [key: string]: any
-    },
+  oidcParameters?: {
+    [key: string]: any
+  },
 }
-interface AppModalLoginMainState {
-    username: string,
-    password: string,
-    failedLogin: boolean,
-    isLoading: boolean,
+
+interface LoginFormValues {
+  username: string;
+  password: string;
 }
-class AppModalLoginMain extends Component<AppModalLoginMainProps, AppModalLoginMainState> {
-  static defaultProps = {
-    onLogin: () => {},
-    onResetPassword: () => {},
-    locationSearchData: '',
-    locationPathName: '',
-    disableLogin: false,
-  };
 
-  constructor(props) {
-    super(props);
+const AppModalLoginMain: React.FC<AppModalLoginMainProps> = (props) => {
+  const {
+    openModal,
+    handleModalClose,
+    onLogin,
+    disableLogin,
+    onResetPassword,
+    showForgotPasswordLink,
+    appModalLoginLinks,
+  } = props;
+
+  const registrationLink = appModalLoginLinks.registration;
+
+  const [failedLogin, setFailedLogin] = useState(false);
+
+  useEffect(() => {
```

Comment:
Do we want to grab this config again on every render?

Reply:
Ah nvm, the empty array implies a render count of only 1, and not "starting at 1". This looks great 😄

Reply:
ah true.  But why is it wrapped in the useEffect() method if we only want to set it once? Couldn't we just place it as a constant at the top of the function?